iot/api/CloudIotTest/CloudIotTest.cs

```diff
@@ -235,46 +262,118 @@ public void TestListGateways()
+                Assert.Contains($"Creating device with id: {deviceId}",
+                    bindDeviceToGatewayOut.Stdout);
                 Assert.Contains("Created device:", bindDeviceToGatewayOut.Stdout);
                 Assert.Contains("Device bound:", bindDeviceToGatewayOut.Stdout);
                 //Check if device is bound to the gateway
-                var listDevicesForGatewayOut = Run("listDevicesForGateway", _fixture.ProjectId, _fixture.RegionId, _fixture.RegistryId, gatewayName);
+                var listDevicesForGatewayOut = Run("listDevicesForGateway", _fixture.ProjectId,
+                    _fixture.RegionId, _fixture.RegistryId, gatewayName);
                 Assert.Contains("Found", listDevicesForGatewayOut.Stdout);
                 Assert.Contains($"ID: {deviceId}", listDevicesForGatewayOut.Stdout);
             }
             finally
             {
-                Run("unbindDeviceFromGateway", _fixture.ProjectId, _fixture.RegionId, _fixture.RegistryId, deviceId, gatewayName);
-                Run("deleteDevice", _fixture.ProjectId, _fixture.RegionId, _fixture.RegistryId, deviceId);
-                Run("deleteDevice", _fixture.ProjectId, _fixture.RegionId, _fixture.RegistryId, gatewayName);
+                Run("unbindDeviceFromGateway", _fixture.ProjectId, _fixture.RegionId,
+                    _fixture.RegistryId, deviceId, gatewayName);
+                Run("deleteDevice", _fixture.ProjectId, _fixture.RegionId,
+                    _fixture.RegistryId, deviceId);
+                Run("deleteDevice", _fixture.ProjectId, _fixture.RegionId,
+                    _fixture.RegistryId, gatewayName);
             }
         }
         //[END iot_gateway_tests]
+
+        //[START iot_gateway_mqtt_tests]
+        [Fact]
+        public void TestSendDataForBoundDevice()
+        {
+            var gatewayId = string.Format("test-gateway-{0}", "RS256");
+            var deviceId = string.Format("test-device-{0}", TestUtil.RandomName());
+
+            //Setup
+            Run("createGateway", _fixture.ProjectId, _fixture.RegionId, _fixture.RegistryId,
+                gatewayId, "test/data/rsa_cert.pem", "RS256");
+            Run("bindDeviceToGateway", _fixture.ProjectId, _fixture.RegionId,
+                _fixture.RegistryId, deviceId, gatewayId);
+
+            //Connect the gateway
+            var sendDataBoundDeviceOut = Run("sendDataFromBoundDevice", _fixture.ProjectId,
+                _fixture.RegionId, _fixture.RegistryId,
+            deviceId, gatewayId, _fixture.PrivateKeyPath, "RS256", _fixture.CertPath,
+            "state", "test-message");
+            Assert.Contains("Data sent", sendDataBoundDeviceOut.Stdout);
+            Assert.Contains("On Publish", sendDataBoundDeviceOut.Stdout);
+
+            Assert.DoesNotContain("An error occured", sendDataBoundDeviceOut.Stdout);
+
+            //Clean up
+            Run("unbindDeviceFromGateway", _fixture.ProjectId, _fixture.RegionId,
```

Comment:
None of this clean up code will run if an assertion fails.  Please add a try {} finally {} block to make sure it always runs.